vortex-array/src/compute/list.rs

```diff
@@ -0,0 +1,127 @@
+//! List-related compute operations.
+
+use arrow_buffer::BooleanBuffer;
+use num_traits::AsPrimitive;
+use vortex_buffer::Buffer;
+use vortex_dtype::{NativePType, match_each_integer_ptype};
+use vortex_error::{VortexResult, vortex_bail};
+use vortex_mask::Mask;
+use vortex_scalar::Scalar;
+
+use crate::arrays::{ConstantArray, ListArray};
+use crate::compute::{Operator, compare};
+use crate::variants::PrimitiveArrayTrait;
+use crate::{Array, ArrayRef, IntoArray, ToCanonical};
+
+/// Get a `Mask` representing the positions in `array` that contains the scalar `value`.
+///
+/// ## Example
+///
+/// ```rust
+/// use vortex_array::{Array, IntoArray};
+/// use vortex_array::arrays::{ListArray, VarBinArray};
+/// use vortex_array::compute::list_contains;
+/// use vortex_array::validity::Validity;
+/// use vortex_buffer::buffer;
+/// use vortex_dtype::DType;
+/// let elements = VarBinArray::from_vec(
+///         vec!["a", "a", "b", "a", "c"], DType::Utf8(false.into())).into_array();
+/// let offsets = buffer![0u32, 1, 3, 5].into_array();
+/// let list_array = ListArray::try_new(elements, offsets, Validity::NonNullable).unwrap();
+///
+/// let matches = list_contains(&list_array, "b".into()).unwrap();
+/// let to_vec: Vec<bool> = matches.to_boolean_buffer().iter().collect();
+/// assert_eq!(to_vec, vec![false, true, false]);
+/// ```
+pub fn list_contains(array: &dyn Array, value: Scalar) -> VortexResult<Mask> {
```

Comment:
We could also add a utility method on `ListArrayTrait` that will call into the top-level function for you.

Might be a good place ultimately for this type-specific compute to live? Another alternative would be to make the receiver here be `&dyn ListArrayTrait`